src/TrimNoOps.cpp

```diff
@@ -222,9 +222,9 @@ class SimplifyUsingBounds : public IRMutator {
                 Scope<Interval> s;
                 // Rearrange the expression if possible so that the
                 // loop var only occurs once.
-                Expr solved = solve_expression(test, loop.var);
-                if (solved.defined()) {
-                    test = solved;
+                SolverResult solved = solve_expression(test, loop.var);
+                if (solved.fully_solved) {
```

Comment:
It might be nice to have a helper function on SolverResult like `if_solved()`, so you could write:

    test = solve_expression(test, loop.var).if_solved();

It seems like this will be a common pattern.